subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -220,6 +220,12 @@ static struct tls_context *tls_alloc(void)
 		mbedtls_pk_init(&tls->priv_key);
 #endif
 
+		/* Set hostname to empty string to enforce it's verification.
+		 * By default, mbedTLS does not verify hostname if it's set to
```

Comment:
This looks like confusion/mixing up different concerns. The purpose of mbedtls_ssl_set_hostname() is to enable use of SNI (TLS VHOST). If mbedtls_ssl_set_hostname() is not called, everything should work normally, except that no SNI extension should be sent. Forcible setting it to empty string to trigger some "hostname verification" seems wrong.